device/core/devdoc/module_client_requirements.md

```diff
@@ -82,8 +82,32 @@ class ModuleClient extends InternalClient {
 
 **SRS_NODE_MODULE_CLIENT_18_022: [** The `sendOutputEventBatch` method shall not throw if the `callback` is not passed. **]**
 
+### setOptions
+**SRS_NODE_MODULE_CLIENT_16_042: [** The `setOptions` method shall throw a `ReferenceError` if the options object is falsy. **]**
 
-**SRS_NODE_MODULE_CLIENT_16_096: [** The `setRetryPolicy` method shall call the `setRetryPolicy` method on the twin if it is set and pass it the `policy` object. **]**
+**SRS_NODE_MODULE_CLIENT_16_043: [** The `done` callback shall be invoked with no parameters when it has successfully finished setting the client and/or transport options. **]**
+
+**SRS_NODE_MODULE_CLIENT_16_044: [** The `done` callback shall be invoked with a standard javascript `Error` object and no result object if the client could not be configured as requested. **]**
+
+**SRS_NODE_MODULE_CLIENT_16_098: [** The `setOptions` method shall call the `setOptions` method with the `options` argument on the `MethodClient` object of the `ModuleClient`. **]**
+
+### invokeMethod(deviceId: string, moduleIdOrMethodParams: string | DirectMethodParams, methodParamsOrCallback: DirectMethodParams | DirectMethodCallback, callback?: DirectMethodCallback): void
+
+**Other valid signatures:**
```

Comment:
Interesting.  So we can call a device method with a module identity?  Is that strange?  Do other SDKs allow this?

Reply:
yes - as long as it's a downstream device. think about the ML module running a workload at the edge on telemetry sent by downstream devices and then triggering something on that device.

---
In reply to: [197223596](https://github.com/Azure/azure-iot-sdk-node/pull/322#discussion_r197223596) [](ancestors = 197223596)